### PR TITLE
docs: format and update developer readme files

### PR DESCRIPTION
**Pull request by NickVolynkin**
Preview:

* https://github.com/NickVolynkin/tarantool/blob/update-dev-readme-files/README.FreeBSD.md
* https://github.com/NickVolynkin/tarantool/blob/update-dev-readme-files/README.MacOSX.md
* https://github.com/NickVolynkin/tarantool/blob/update-dev-readme-files/README.OpenBSD.md

Merge https://github.com/tarantool/doc/pull/2698 after this PR

### CODE REVIEW

File: README.MacOSX.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+make
+```
+
+In this example, we are making a developer's build (`-DCMAKE_BUILD_TYPE=RelWithDebInfo`)
+with cmake using MacPorts (`-DDARWIN_BUILD_TYPE=Ports`). To build a release version,
+set `-DCMAKE_BUILD_TYPE=Release`.
+
+Remember also to set up the cmake's flag `-DDARWIN_BUILD_TYPE` depending on the package
+manager you use (`'-DDARWIN_BUILD_TYPE=None'` for Xcode Tools and Homebrew, and
+`'-DDARWIN_BUILD_TYPE=Ports'` for MacPorts). It is set to `None` by default.
```
Comment (Totktonada) [inline README.MacOSX.md:108]: Quotes are redundant, when backticks are used.

File: README.FreeBSD.md [hunk 2]
```diff
@@ -1,44 +1,52 @@
 Target OS: FreeBSD 12.2 (RELEASE) and FreeBSD 13.0 (RELEASE)
 
-1. Install necessary packages:
--------------
-pkg install git cmake gmake readline icu libiconv
+# 1. Install necessary packages:
```
[the remark this answers is not in the document]
Comment (NickVolynkin) [reply README.FreeBSD.md]: fixed

File: README.MacOSX.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+-DCMAKE_BUILD_TYPE=RelWithDebInfo \
+-DCURL_INCLUDE_DIR=$(xcode-select --sdk macosx --show-sdk-path)/usr/include \
+-DCURL_LIBRARY=/usr/lib/libcurl.dylib \
+-DDARWIN_BUILD_TYPE=Ports
+make
+```
+
+In this example, we are making a developer's build (`-DCMAKE_BUILD_TYPE=RelWithDebInfo`)
+with cmake using MacPorts (`-DDARWIN_BUILD_TYPE=Ports`). To build a release version,
+set `-DCMAKE_BUILD_TYPE=Release`.
```
[the remark this answers is not in the document]
Comment (NickVolynkin) [reply README.MacOSX.md]: I've noted that Release isn't supported for macOS.

File: README.MacOSX.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+For the default Xcode Tools by Apple:
+
+```bash
+sudo xcode-select --install
+sudo xcode-select -switch /Applications/Xcode.app/Contents/Developer
+```
+
+For Homebrew:
+```bash
+brew install openssl readline curl icu4c libiconv zlib cmake python@3.8
```
[the remark this answers is not in the document]
Comment (Totktonada) [reply README.MacOSX.md:27]: Actual changes are still mixed with formatting changes. We don't act this way.

Also I don't understand why you silently resolved this comment. It is obviously not fixed.

**Comment by NickVolynkin** [comment]
@TarantoolBot document
test comment creation

**Comment by NickVolynkin** [comment]
@TarantoolBot document
Title: test title
test issue creation

**Pull request closed without merging**